### PR TITLE
Updating go to v1.20

### PR DESCRIPTION
**Pull request by marosset**
/kind cleanup

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> 
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Updating golang to v1.20
```

### CODE REVIEW

**Comment by mauriciopoppe** [comment]
Presubmit failed with:
```
{  
Repo uses 'go mod'.
+ env GO111MODULE=on go mod tidy
go: go.mod file indicates go 1.20, but maximum version supported by tidy is 1.19
ERROR: vendor check failed.
    }
```

The go version is coming from release-tools, I think we need to sync that repo here before doing this change

**Comment by mauriciopoppe** [comment]
We updated release-tools in https://github.com/kubernetes-csi/csi-proxy/pull/290, please rebase when you get the chance.

There are also some codespell false negatives:

```
Resulting CLI options  --check-filenames --skip ./.git,./.github/workflows/codespell.yml,.git,*.png,*.jpg,*.svg,*.sum,./vendor,go.sum,./release-tools/prow.sh
3
Error: ./integrationtests/csi_api_gen_test.go:19: bootstraping ==> bootstrapping
Error: ./release-tools/contrib/get_supported_version_csi-sidecar.py:35: ouputs ==> outputs
Error: ./release-tools/contrib/get_supported_version_csi-sidecar.py:101: relase ==> release
```

Could you please also add `release-tools/contrib/*.py` to https://github.com/kubernetes-csi/csi-proxy/blob/fb21c30025296703e21fec7fc50c9e3a80eca86f/.github/workflows/codespell.yml#L15.

There's also this failure:

```
Run go get -u github.com/golang/protobuf/{proto,protoc-gen-go}
go: go.mod file not found in current directory or any parent directory.
	'go get' is no longer supported outside a module.
	To build and install a command, use 'go install' with a version,
	like 'go install example.com/cmd@latest'
	For more information, see https://golang.org/doc/go-get-install-deprecation
	or run 'go help get' or 'go help install'.
Error: Process completed with exit code 1.
```

We need to update a few scripts that were doing `go get` to go install.

Sorry about all of these failures! We haven't received a PR in a long time for this project so it got out of date, if you don't have the bandwidth then we can take it, just let us know.

**Comment by marosset** [comment]
> er



> We updated release-tools in #290, please rebase when you get the chance.
> 
> There are also some codespell false negatives:
> 
> ```
> Resulting CLI options  --check-filenames --skip ./.git,./.github/workflows/codespell.yml,.git,*.png,*.jpg,*.svg,*.sum,./vendor,go.sum,./release-tools/prow.sh
> 3
> Error: ./integrationtests/csi_api_gen_test.go:19: bootstraping ==> bootstrapping
> Error: ./release-tools/contrib/get_supported_version_csi-sidecar.py:35: ouputs ==> outputs
> Error: ./release-tools/contrib/get_supported_version_csi-sidecar.py:101: relase ==> release
> ```
> 
> Could you please also add `release-tools/contrib/*.py` to
> 
> https://github.com/kubernetes-csi/csi-proxy/blob/fb21c30025296703e21fec7fc50c9e3a80eca86f/.github/workflows/codespell.yml#L15
> 
> .
> There's also this failure:
> 
> ```
> Run go get -u github.com/golang/protobuf/{proto,protoc-gen-go}
> go: go.mod file not found in current directory or any parent directory.
> 	'go get' is no longer supported outside a module.
> 	To build and install a command, use 'go install' with a version,
> 	like 'go install example.com/cmd@latest'
> 	For more information, see https://golang.org/doc/go-get-install-deprecation
> 	or run 'go help get' or 'go help install'.
> Error: Process completed with exit code 1.
> ```
> 
> We need to update a few scripts that were doing `go get` to go install.
> 
> Sorry about all of these failures! We haven't received a PR in a long time for this project so it got out of date, if you don't have the bandwidth then we can take it, just let us know.

No problem.
I also just submitted a fix into csi-release-tools to fix the codespell errors.

@mauriciopoppe - if you have bandwidth to address to `go get` installs I'd appreciate it!

**Comment by mauriciopoppe** [comment]
> I also just submitted a fix into csi-release-tools to fix the codespell errors.

Thanks, it takes some time to bring that repo into all of the CSI repos so we might need to make that change in this repo too.

> if you have bandwidth to address to go get installs I'd appreciate it!

Sure, I'll work on both codespell/go get in a new commit

File: go.mod [hunk 1]
```diff
@@ -24,12 +24,23 @@ require (
...
-	k8s.io/klog v1.0.0 // indirect
 	k8s.io/klog/v2 v2.9.0
 )
+
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/go-logr/logr v0.4.0 // indirect
+	github.com/golang/protobuf v1.4.3 // indirect
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	golang.org/x/net v0.0.0-20200202094626-16171245cfb2 // indirect
```
[the remark this answers is not in the document]
Comment (mauriciopoppe) [reply go.mod:39]: sure, fixed in https://github.com/kubernetes-csi/csi-proxy/pull/292

**Comment by mauriciopoppe** [comment]
I made the go 1.20 change too in https://github.com/kubernetes-csi/csi-proxy/pull/292

**Comment by marosset** [comment]
> I made the go 1.20 change too in #292

thanks @mauriciopoppe!

**Pull request closed without merging**